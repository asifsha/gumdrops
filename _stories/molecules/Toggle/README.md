`<Toggle>` components come in two flavors: radio and checkbox. Each one behaves in the same manner as their respective standard input type.

**NOTE**: This component can be controlled or uncontrolled. This component accepts any prop that you may want to add, which are applied to the input. The only drawback is that there is not validation for unlisted props.  **It is highly recommended that you use controlled components in React.**

*Recommended Props*: `name`, `checked`, `defaultChecked`, `onChange`

**Controlled example - Example for single Toggle**:
```
state = {
    myToggle: false
}

_handleToggle = () => {
    this.setState(({ myToggle }) => ({ myToggle: !myToggle }));
}

return (
    <Toggle
        name="myToggle"
        label="Default Toggle"
        checked={ this.state.myToggle }
        onChange={ this._handleToggle }
    />
)
```

**Uncontrolled example**:
```
_handleSubmit = (e) => {
    e.preventDefault();
    const { myToggle } = e.target.elements;
    console.log(myToggle.checked);
};

render() {
    return (
        <form onSubmit={ this._handleSubmit }>
            <FormGroup>
                { /* Pass defaultChecked if you want the input to be checked initially */ }
                <Toggle
                    name="myToggle"
                    label="Default Toggle"
                    defaultChecked
                />
            </FormGroup>
        </form>
    );
}
```
