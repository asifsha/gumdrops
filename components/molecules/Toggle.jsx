import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const Toggle = ({ type, size, label, style, className, ...otherProps }) => {
    const baseClass = 'gds-form-group__toggleswitch';

    const rootClass = cx(baseClass, className, {
        [`${baseClass}--${size}`]: size
    });

    return (
        <div className={rootClass} style={style}>
            <label className="gds-form-group__toggleswitch-label">
                <input className="gds-form-group__toggleswitch-input" type={type} {...otherProps} />
                <span className="gds-form-group__toggleswitch-indicator" />
                {label}
            </label>
        </div>
    );
};

Toggle.displayName = 'Toggle';

Toggle.defaultProps = {
    type: 'checkbox',
    label: ''
};

Toggle.propTypes = {
    /** radio, checkbox */
    type: PropTypes.oneOf(['radio', 'checkbox']),
    label: PropTypes.string,
    /** xs, sm */
    size: PropTypes.string,
    style: PropTypes.object,
    className: PropTypes.string
};

export default Toggle;
