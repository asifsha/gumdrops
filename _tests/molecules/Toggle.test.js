/* globals mount */
import React from 'react';
import Toggle from '../../components/molecules/Toggle';

const defaultProps = {
    type: 'checkbox',
    label: 'Toggle Label',
    size: 'sm',
    style: { width: 100 },
    className: 'custom-class'
};

describe('Expect <Toggle>', () => {
    it('to render', () => {
        const wrapper = mount(<Toggle {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });
});
