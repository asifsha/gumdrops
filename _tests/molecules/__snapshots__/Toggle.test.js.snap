// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Toggle> to render 1`] = `
<Toggle
  className="custom-class"
  label="Toggle Label"
  size="sm"
  style={
    Object {
      "width": 100,
    }
  }
  type="checkbox"
>
  <div
    className="gds-form-group__toggleswitch custom-class gds-form-group__toggleswitch--sm"
    style={
      Object {
        "width": 100,
      }
    }
  >
    <label
      className="gds-form-group__toggleswitch-label"
    >
      <input
        className="gds-form-group__toggleswitch-input"
        type="checkbox"
      />
      <span
        className="gds-form-group__toggleswitch-indicator"
      />
      Toggle Label
    </label>
  </div>
</Toggle>
`;
